src/api/java/io/github/opencubicchunks/cubicchunks/api/worldgen/CubeGeneratorsRegistry.java

```diff
@@ -54,10 +58,31 @@
 
 public class CubeGeneratorsRegistry {
 
+    /** Provide mapping for possible vanilla compatibility generators options */
+    private static final Map<ResourceLocation,BiFunction<IChunkGenerator, World, ICubeGenerator>> vanillaCompatibilityGenerators = new HashMap<ResourceLocation,BiFunction<IChunkGenerator, World, ICubeGenerator>>();
+    /** Provide mapping for vanilla compatibility generators names */
+    private static final Map<ResourceLocation,String> vanillaCompatibilityGeneratorsUnlocalisedNames = new HashMap<ResourceLocation,String>();
+    
     /** List of populators added by other mods to vanilla compatibility generator type */
     private static final List<ICubicPopulator> customPopulatorsForFlatCubicGenerator = new ArrayList<ICubicPopulator>();
 
     private static TreeSet<GeneratorWrapper> sortedGeneratorList = new TreeSet<>();
+    public static final ResourceLocation defaultCompatibilityGenerator = new ResourceLocation("cubicchunks","default");
+
+    /**
+     * Register a compatibility generator - something that will convert vanilla
+     * generators into cubic form
+     *
+     * @param unLocalisedName - GUI name
+     * @param name of a generator
+     * @param generatorProvider - BiFunction which will consume vanilla chunk
+     *        generator and world instances and return cubic generator
+     */
+    public static void register(String unLocalisedName, ResourceLocation name, BiFunction<IChunkGenerator, World, ICubeGenerator> generatorProvider) {
```

Comment:
Make it a forge registry. Loading a world with some of the compatibility generators missing is dangerous and forge should warn against it.